compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion/Primitives.hs

```diff
@@ -254,13 +254,108 @@ convertPrim _ "BETextReplicate" (TInt64 :-> TText :-> TText) = EBuiltin BETextRe
 convertPrim _ "BETextSplitOn" (TText :-> TText :-> TList TText) = EBuiltin BETextSplitOn
 convertPrim _ "BETextIntercalate" (TText :-> TList TText :-> TText) = EBuiltin BETextIntercalate
 
+-- Template Desugaring.
+
+convertPrim _ "UCreate" (TCon template :-> TUpdate (TContractId (TCon template')))
+    | template == template' =
+    -- TODO: restrict to known templates
```

Comment:
Given that `primitive` is only something that we use internally and this would be caught by the LF typechecker I don’t think we ever need to try to catch this here. Same for all the other TODOs here.

Reply:
Hmm, makes sense, but I guess `ToAnyTemplate`/`FromAnyTemplate`/`ToAnyChoice`/... should still have checks? Because the typechecker isn't going to enforce anything there (it just seems `EFromAny` and `EToAny`).

Reply:
good point I forgot about those! That said, I’m fine leaving this for a separate PR. This is still internal and not something we expose our users to.

Reply:
sounds good :)